### PR TITLE
OpenAPI specifications for POST status/messages

### PR DESCRIPTION
**Pull request by rubhanazeem**
Specifications for POST status/messages

### CODE REVIEW

File: src/api/public/apidocs-new/paths/status_messages.yaml [hunk 1]
```diff
@@ -30,3 +30,31 @@ get:
...
+  requestBody:
+    description: Attributes for status message
+    content:
+      application/xml; charset=utf-8:
+        schema:
+          $ref: '../components/schemas/status_message.yaml'
+        example:
+          message: Feature number 42 was implemented!
+          severity: announcement
+          scope: all_users
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/status_messages.yaml:51]:
```suggestion
  requestBody:
    description: Attributes for status message
    required: true
    content:
      application/xml; charset=utf-8:
        schema:
          $ref: '../components/schemas/status_message.yaml'
        example:
          message: Feature number 42 was implemented!
          severity: announcement
          scope: all_users
```
This looks like the request body should be required.

File: src/api/public/apidocs-new/paths/status_messages.yaml [hunk 2]
```diff
@@ -30,3 +30,31 @@ get:
...
+  responses:
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/status_messages.yaml:52]: We should also add an error response when the request body is empty. This is what I tried and what I got in my development environment:
```bash
> curl -u Admin:opensuse -H 'Content-Type: application/xml; charset=utf-8' -X POST "http://localhost:3000/status/messages"
<status code="validation_failed">
  <summary>Document is empty, not allowed for status_message</summary>
</status>

```

File: src/api/public/apidocs-new/paths/status_messages.yaml [hunk 3]
```diff
@@ -30,3 +30,31 @@ get:
       $ref: '../components/responses/unauthorized.yaml'
   tags:
     - Status messages
+
+post:
+  summary: Create a status message
+  description: |
+    Create a new status message (Only admins are allowed to add status messages).
+    Possible values for severity: (information, green, yellow, red, announcement).
+    Possible values for scope: (all_users, logged_in_users, admin_users, in_beta_users, in_rollout_users)
```
[the remark this answers is not in the document]
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/status_messages.yaml]: Agree. It's updated

File: src/api/public/apidocs-new/paths/status_messages.yaml (hunk 2 above)
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/status_messages.yaml:55]: I've added a missing 400 error

File: src/api/public/apidocs-new/components/responses/bad_request_error.yaml [hunk 4]
```diff
@@ -14,3 +14,7 @@ content:
         value:
           code: err_register_save
           summary: "Could not save the registration, details: Login is the name of an already existing user and Password can't be blank."
+      Validation Failed:
+        value:
+          code: validation_failed
+          summary: "status_message validation error: 2:0: ERROR: Element '<some element>': No matching global declaration available for the validation root."
```
Comment (eduardoj) [inline src/api/public/apidocs-new/components/responses/bad_request_error.yaml:20]: The component `responses/bad_request_error.yaml` is also used in the `POST /person/{login}` endpoint.

Does this "Validation Failed" error introduced for status messages apply also for the person endpoint? If not, I wouldn't add the "Validation Failed" example into the `responses/bad_request_error.yaml` component.

**Review by eduardoj — APPROVED** [review]

**Pull request merged**